### PR TITLE
Explain why client stub mocking is discouraged

### PR DESCRIPTION
**Pull request by rmichela**
As seen in hacker news comment https://news.ycombinator.com/item?id=20025085 of story https://news.ycombinator.com/item?id=20021759.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -107,6 +107,16 @@ We encourage users to leverage `InProcessTransport` as demonstrated in the examp
 write unit tests. `InProcessTransport` is light-weight and runs the server
 and client in the same process without any socket/TCP connection.
 
+Mocking the client stub provides a false sense of security when writing tests. Mocking stubs and responses allows for tests
+that don't map to reality, causing the tests to pass, but the system-under-test to fail. Example bugs not caught by mocked 
+stub tests include:
+
+* Calling the stub with a NULL message
```
Comment (carl-mastrangelo) [inline examples/README.md]: `null` ?

File: examples/README.md [hunk 2]
```diff
@@ -107,6 +107,16 @@ We encourage users to leverage `InProcessTransport` as demonstrated in the examp
 write unit tests. `InProcessTransport` is light-weight and runs the server
 and client in the same process without any socket/TCP connection.
 
+Mocking the client stub provides a false sense of security when writing tests. Mocking stubs and responses 
+allows for teststhat don't map to reality, causing the tests to pass, but the system-under-test to fail.
```
Comment (carl-mastrangelo) [inline examples/README.md]: s/teststhat/tests that

File: examples/README.md [hunk 3]
```diff
@@ -107,6 +107,16 @@ We encourage users to leverage `InProcessTransport` as demonstrated in the examp
 write unit tests. `InProcessTransport` is light-weight and runs the server
 and client in the same process without any socket/TCP connection.
 
+Mocking the client stub provides a false sense of security when writing tests. Mocking stubs and responses 
+allows for tests that don't map to reality, causing the tests to pass, but the system-under-test to fail.
```
Comment (zhangkun83) [inline examples/README.md]: I would also mention that it's very hard to accurately reproduce the reality by mocks, due to the complexity of the gRPC client library.

**Comment by zhangkun83** [comment]
One suggested addition. Otherwise LGTM.

File: examples/README.md [hunk 4]
```diff
@@ -107,8 +107,11 @@ We encourage users to leverage `InProcessTransport` as demonstrated in the examp
 write unit tests. `InProcessTransport` is light-weight and runs the server
 and client in the same process without any socket/TCP connection.
 
-Mocking the client stub provides a false sense of security when writing tests. Mocking stubs and responses 
+Mocking the client stub provides a false sense of security when writing tests. Mocking stubs and responses
 allows for tests that don't map to reality, causing the tests to pass, but the system-under-test to fail. 
+The gRPC client library is complicated, and accurately reproducing that complexity with mocks very hard.
```
Comment (zhangkun83) [inline examples/README.md]: .. **is** very hard.

Comment (rmichela) [reply examples/README.md]: Fixed.

File: examples/README.md (hunk 3 above)
Comment (rmichela) [reply examples/README.md]: Fixed

**Pull request merged**